spec/inject_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'array'
+
+describe Array do
+
+let(:array) {Array.new}
+
+		it 'can take an argument' do 
+			expect(array).to receive(:my_inject).with(0)
+			array.my_inject(0)
+		end 
+
+		# it 'can take a block' do
+		# 	expect(array).to receive(:my_inject).with({"this is my block"})
+		# 	array.my_inject(0){"this is my block"}
+		# end
+
```

Comment:
You can pass a block as an argument by using the `&` prefix - so `[1, 2, 3, 4, 5].inject(&:+)`, for example, or a more complex block